VideoIndexerEnabledByArc/scripts/vi_extension_install.sh

```diff
@@ -0,0 +1,203 @@
+      echo "Max number of retries reached without getting Cognitive Service Secrets . Exiting script."
+      printf "\n"
+      exit 1
+    fi
+  done
+
+}
+##################################################################
+# Create Cognitive Services HOBO (On Behalf Of the User ) Resources
+##################################################################
+function create_cognitive_hobo_resources {
+  echo -e "\t create Cognitive Services On VI RP ***start***"
+  sleepDuration=10
+  createResourceUri="https://management.azure.com/subscriptions/${subscriptionId}/resourceGroups/${resourceGroup}/providers/Microsoft.VideoIndexer/accounts/${accountName}/CreateExtensionDependencies?api-version=${viApiVersion}"
+  responseString=$(az rest --method post --uri $createResourceUri --verbose 2>&1 >/dev/null || true)
+  responseStatusLine=$(echo "$responseString" | grep 'INFO: Response status:')
+  responseStatus=$(echo "$responseStatusLine" | grep -oP '\d+')
+  responseCode=$((responseStatus))
+
+  if [[ "$responseCode" == 202 ]]; then
+    echo "Cognitive Services resources are being created. Waiting for completion"
+  elif [[ "$responseCode" == 409 ]]; then
+    echo "Cognitive Services resources already exist. Moving on."
+  else
+    echo "an Unknown error occured: $responseStatus . Exiting"
+    exit 1
+  fi
+
+  get_cs_auth_credentials
+  echo -e "\t create Cognitive Services On VI RP ***done***"
+}
+
+#===========================================================================================================#
+# @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ Main Script @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+#===========================================================================================================#
+
+#=============================================#
+#============== Constants           ==========#
+#=============================================#
+aksVersion="1.28.2"               # https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli
+viApiVersion="2023-06-02-preview" # VI API version
+
+#=============================================#
+#============== Parameters Customization =====#
+#=============================================#
+region="eastus"                     #default region , will be override later
+extensionName="video-indexer"       #default extension name , will be override later
+resourcesPrefix="vi"                #default resources prefix , will be override later
+namespace="video-indexer"           #default namespace , will be override later
+ENDPOINT_URI="127.0.0.1/vi" #default endpoint uri
+version="1.0.41"                    #VI Extension version
+singleNodeStorageClass="local-path" #default storage class for single node cluster
+singleNodeAccessMode="ReadWriteOnce" #default access mode for single node cluster
+region="eastus"
+extensionName="vi-arc"
+namespace="video-indexer"
+
+subscriptionId=""
+resourceGroup=""
+accountName=""
```

Comment:
Is it still needed? Did you contact arc support/office hours to figure it out?

Reply:
not yet